pkgs/development/haskell-modules/configuration-nix.nix

```
@@ -870,6 +870,9 @@ self: super: builtins.intersectAttrs super {
     hinotify = if pkgs.stdenv.hostPlatform.isLinux then self.hinotify else self.fsnotify;
   });
 
+  # needs git to configure
+  git-repair = addBuildTool pkgs.git super.git-repair;
```

We need to be mindful of cross compilation.

```suggestion
  git-repair = addBuildTool pkgs.buildPackages.git super.git-repair;
```

Alternatively set `FORCE_GIT_VERSION=${pkgs.git.version}` in `preConfigure`. That would work without ever executing `git`.